src/Fmt_ast.ml

```diff
@@ -170,329 +170,11 @@ let fmt_recmodule c ctx items f ast =
-                Cmts.relocate c.cmts ~src ~before:e1.pexp_loc ~after ) ) ;
-        (xop, [(l1, sub_exp ~ctx e1)]) :: op_args2
-    | _ -> [(xop, [xexp])]
-  in
-  sugar_infix_ None ~relocate:false (Nolabel, xexp)
-
-let rec sugar_list_pat c pat =
-  let ctx = Pat pat in
-  let {ppat_desc; ppat_loc= src} = pat in
-  match ppat_desc with
-  | Ppat_construct ({txt= Lident "[]"; loc}, None) ->
-      Cmts.relocate c.cmts ~src ~before:loc ~after:loc ;
-      Some ([], loc)
-  | Ppat_construct
-      ( {txt= Lident "::"; loc}
-      , Some {ppat_desc= Ppat_tuple [hd; tl]; ppat_loc; ppat_attributes= []}
-      ) -> (
-    match sugar_list_pat c tl with
-    | Some (xtl, nil_loc) when List.is_empty tl.ppat_attributes ->
-        Some (([src; loc; ppat_loc], sub_pat ~ctx hd) :: xtl, nil_loc)
-    | _ -> None )
-  | _ -> None
-
-let sugar_list_exp c exp =
-  let rec sugar_list_exp_ exp =
-    let ctx = Exp exp in
-    let {pexp_desc; pexp_loc= src} = exp in
-    match pexp_desc with
-    | Pexp_construct ({txt= Lident "[]"; loc}, None) ->
-        Cmts.relocate c.cmts ~src ~before:loc ~after:loc ;
-        Some ([], loc)
-    | Pexp_construct
-        ( {txt= Lident "::"; loc}
-        , Some
-            {pexp_desc= Pexp_tuple [hd; tl]; pexp_loc; pexp_attributes= []}
-        ) -> (
-      match sugar_list_exp_ tl with
-      | Some (xtl, nil_loc) when List.is_empty tl.pexp_attributes ->
-          Some (([src; loc; pexp_loc], sub_exp ~ctx hd) :: xtl, nil_loc)
-      | _ -> None )
-    | _ -> None
-  in
-  let r = sugar_list_exp_ exp in
-  assert (Bool.equal (Option.is_some r) (is_sugared_list exp)) ;
-  r
-
-let sugar_infix_cons xexp =
-  let rec sugar_infix_cons_ ({ast= exp} as xexp) =
-    let ctx = Exp exp in
-    let {pexp_desc; pexp_loc= l1} = exp in
-    match pexp_desc with
-    | Pexp_construct
-        ( {txt= Lident "::"; loc= l2}
-        , Some
-            { pexp_desc= Pexp_tuple [hd; tl]
-            ; pexp_loc= l3
-            ; pexp_attributes= [] } ) -> (
-      match sugar_infix_cons_ (sub_exp ~ctx tl) with
-      | xtl when List.is_empty tl.pexp_attributes ->
-          ([l1; l2; l3], sub_exp ~ctx hd) :: xtl
-      | _ -> [([l1; l2; l3], sub_exp ~ctx hd); ([], sub_exp ~ctx tl)] )
-    | _ -> [([], xexp)]
-  in
-  sugar_infix_cons_ xexp
-
-let rec sugar_ite c ({ast= exp} as xexp) =
-  let ctx = Exp exp in
-  let {pexp_desc; pexp_loc; pexp_attributes} = exp in
-  match pexp_desc with
-  | Pexp_ifthenelse (cnd, thn, Some els) ->
-      Cmts.relocate c.cmts ~src:pexp_loc ~before:cnd.pexp_loc
-        ~after:els.pexp_loc ;
-      (Some (sub_exp ~ctx cnd), sub_exp ~ctx thn, pexp_attributes)
-      :: sugar_ite c (sub_exp ~ctx els)
-  | Pexp_ifthenelse (cnd, thn, None) ->
-      Cmts.relocate c.cmts ~src:pexp_loc ~before:cnd.pexp_loc
-        ~after:thn.pexp_loc ;
-      [(Some (sub_exp ~ctx cnd), sub_exp ~ctx thn, pexp_attributes)]
-  | _ -> [(None, xexp, pexp_attributes)]
-
 let sugar_sequence c width xexp =
```

Comment:
It is odd to leave this small piece and move the rest. Can you explain?

Reply:
I meant that it is probably better to inline this little snippet into the couple places it is used, as all the other similar calls of `List.group`.

Reply:
Done.